protocols/v2/subprotocols/mining/src/lib.rs

```diff
@@ -356,13 +426,210 @@ impl ExtendedExtranonce {
         }
     }
 }
-
+// This function is used to inctrement extranonces, and it is used in next_standard and
+// next_extended functions. Returns Err(()) if the the input an array of MAX.
 fn increment_bytes_be(bs: &mut [u8]) -> Result<(), ()> {
     for b in bs.iter_mut().rev() {
         if *b != u8::MAX {
             *b += 1;
             return Ok(());
+        } else {
+            *b = 0;
         }
     }
+    for b in bs.iter_mut() {
```

Comment:
To get to this loop `bs` must have all u8::MAX elements - which will be set to 0 by the above loop - and then in this loop every element will be set back to u8::MAX - is that really what you wanted? Then if it's passed in again it will continually have every element set to 0 and then to u8::MAX.